ride_share.rb

```diff
@@ -0,0 +1,108 @@
+
+drivers = {
+  DR0001: [
+    {
+      date: "3rd Feb 2016",
+      cost: 10,
+      rider: "RD0003", 
+      rating: 3
+    }, 
+    {
+      date: "3rd Feb 2016", 
+      cost: 30, 
+      rider: "RD0015",
+      rating: 4 
+    }, 
+    {
+      date: "5th Feb 2016",
+      cost: 45,
+      rider: "RD0003",
+      rating: 2
+    },
+  ],
+  DR0002: [
+    {
+      date: "3rd Feb 2016",
+      cost: 25,
+      rider: "RD0073",
+      rating: 5
+    },
+    {
+      date: "4th Feb 2016",
+      cost: 15,
+      rider: "RD0013",
+      rating: 1
+    },
+    {
+      date: "5th Feb 2016",
+      cost: 35,
+      rider: "RD0066",
+      rating: 3
+    }
+  ],
+  DR0003: [
+    {
+      date: "4th Feb 2016",
+      cost: 5,
+      rider: "RD0066",
+      rating: 5
+    },
+    {
+      date: "5th Feb 2016",
+      cost: 50,
+      rider: "RD0003",
+      rating: 2
+    }
+  ],
+  DR0004: [
+    {
+      date: "3rd Feb 2016",
+      cost: 5,
+      rider: "RD0022",
+      rating: 5
+    },
+    {
+      date: "4th Feb 2016",
+      cost: 10,
+      rider: "RD0022",
+      rating: 4
+    },
+    {
+      date: "5th Feb 201",
+      cost: 20,
+      rider: "RD0073",
+      rating: 5
+    }
+  ]
+}
+
+
+high_rating = 0
+high_earn = 0
+driver_rating = ""
+driver_earn = ""
+
+
+drivers.each do |key, value|
```

Comment:
Hm, may I suggest renaming these variables to `driver` and `trips`, because the `key` represents the driver, and the `value` represents an array of trips they've driven?